src/Sylius/Bundle/ArchetypeBundle/Form/EventListener/ParentArchetypeListener.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ArchetypeBundle\Form\EventListener;
+
+use Sylius\Component\Archetype\Model\ArchetypeInterface;
+use Symfony\Component\Form\Exception\UnexpectedTypeException;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+
+/**
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ */
+class ParentArchetypeListener implements EventSubscriberInterface
+{
+    /**
+     * @var string
+     */
+    private $subject;
+
+    /**
+     * @param string $subject
+     */
+    public function __construct($subject)
+    {
+        $this->subject = $subject;
+    }
+
+    /**
+     * @return array
+     */
+    public static function getSubscribedEvents()
+    {
+        return array(FormEvents::PRE_SET_DATA => 'preSetData');
+    }
+
+    /**
+     * @param FormEvent $event
+     */
+    public function preSetData(FormEvent $event)
+    {
+        $currentArchetype = $event->getData();
+        if (!$currentArchetype instanceof ArchetypeInterface) {
+           throw new UnexpectedTypeException($currentArchetype, 'Sylius\Component\Archetype\Model\ArchetypeInterface');
+        }
+
+        $form = $event->getForm();
+        $parentOptions = array(
+            'required' => false,
+            'label' => 'sylius.form.archetype.parent',
+            'query_builder' => function (RepositoryInterface $repository) use ($currentArchetype) {
+                return $repository
+                    ->createQueryBuilder('parentArchetype')
```

Comment:
It is not really ``parentArchetype``, we are selecting all archetypes except current one. ``archetype`` will be enough or you can actually just use ``o`` as alias, cause that will be new convention for query builders.